src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -1223,6 +1230,120 @@ export default class SourceFileLinter {
 		}
 	}
 
+	#analyzeModelDataTypes(node: ts.NewExpression | ts.CallExpression) {
+		node.arguments?.forEach((arg) => {
+			// Only handle object literals, ignoring the optional first id argument or other unrelated arguments
+			if (!ts.isObjectLiteralExpression(arg)) {
+				return;
+			}
+			arg.properties.forEach((prop) => {
+				if (!ts.isPropertyAssignment(prop)) {
+					return;
+				}
+
+				// Get the type property
+				let typeField;
+				if (ts.isObjectLiteralExpression(prop.initializer)) {
+					typeField = prop.initializer.properties.find((prop: ts.ObjectLiteralElementLike) => {
+						return ts.isPropertyAssignment(prop) &&
+							ts.isIdentifier(prop.name) &&
+							prop.name.text === "type";
+					});
+				} else if (ts.isStringLiteral(prop.initializer) &&
+					ts.isIdentifier(prop.name) && prop.name.text === "type") {
+					typeField = prop;
+				}
+
+				if (typeField && ts.isPropertyAssignment(typeField) &&
+					(!ts.isNewExpression(node) /* bindProperty() */ ||
+						this.#isPropertyBindingType(node, prop.name.getText()) /* new Control() */)) {
+					this.#analyzeModelTypeField(typeField.initializer);
+				} else if (ts.isStringLiteral(prop.initializer) &&
+					prop.initializer.text.startsWith("{") && prop.initializer.text.endsWith("}") &&
+					ts.isNewExpression(node) &&
+					this.#isPropertyBindingType(node, prop.name.getText())) {
+					const originalFilename = Array.from(
+						this.context.getMetadata(this.resourcePath)?.xmlCompiledResources ?? [])[0];
+					let resourcePath = this.resourcePath;
+					if (originalFilename) {
+						if ([".view.xml", ".fragment.xml"].some((ending) => originalFilename.endsWith(ending))) {
+							// Do not process xml-s. This case would be handled separately withing the BindingParser
+							return;
+						}
+						resourcePath = resourcePath.split("/").splice(0, -1).join("/") + "/" + originalFilename;
+					}
+
+					const imports = this.sourceFile.statements
+						.filter((stmnt): stmnt is ts.ImportDeclaration =>
+							stmnt.kind === ts.SyntaxKind.ImportDeclaration)
+						.map((importNode) => {
+							return {
+								moduleName: (importNode.moduleSpecifier as ts.StringLiteral).text,
+								variableName: importNode.importClause?.name?.text,
+							} as RequireDeclaration;
+						});
+
+					const nodeSourceMap = this.sourceMaps?.get(this.resourcePath);
+					const traceMap = nodeSourceMap ? new TraceMap(nodeSourceMap) : undefined;
+					const {start: nodePos} = getPositionsForNode({
+						node: prop,
+						sourceFile: this.sourceFile,
+						resourcePath: this.resourcePath,
+						traceMap: traceMap,
+					});
```

Comment:
As we won't check code that is transpiled from XML, I wouldn't expect that this special handling is needed at all.